internal/pkg/util/host.go

```diff
@@ -31,3 +33,61 @@ func GetHostName(isFqdnFeatureEnabled bool, hostInfo types.HostInfo, host types.
 
 	return fqdn
 }
+
+var _ types.Host = &threadSafeHost{}
+
+// threadSafeHost is a thread-safe wrapper around types.Host.
+// It exists so we can only create it once, as some of the setup it does is relatively expensive.
+type threadSafeHost struct {
+	sync.Mutex
+	inner types.Host
+}
+
+func newThreadSafeHost(inner types.Host) *threadSafeHost {
+	return &threadSafeHost{inner: inner}
+}
+
+func (s *threadSafeHost) CPUTime() (types.CPUTimes, error) {
+	s.Lock()
```

Comment:
my first reaction/thinking was that this thread-safety code most probably belongs to the go-sysinfo codebase?! but then looking again in the exposed functions, and from a quick look in go-sysinfo they do not mutate anything, so I am wondering do we need to hold a mutex to call them?! did I miss a mutation?

Reply:
go-sysinfo doesn't guarantee that these are thread-safe, so I don't have much of a choice here. We could try to make them so in the library, but I'm not sure it's worth it. These functions aren't called very often, so in practice there shouldn't be any contention.

Reply:
since these functions are not mutating anything I would propose to remove the mutex. Or again you may have spotted a mutation that I missed?! If not, I am not entirely convinced that losing the concurrency of calling these functions, given that this is now a shared instance across multiple places, is wise. But if you insist sure go with it 🙂

Reply:
The functions are not mutating anything *right now*, but go-sysinfo doesn't guarantee this interface is thread-safe, so this is an implementation detail that may change. If we want to remove the locks, we should make the change in go-sysinfo, and I don't think that's worth the hassle. This PR is a bit borderline already in my opinion when it comes to complexity introduced vs performance gained.

At the very least, I'd have to see evidence of actual contention before looking into this further.